app/controllers/actions/IdentifierAction.scala

```diff
@@ -39,47 +40,122 @@ trait IdentifierAction extends ActionBuilder[IdentifierRequest, AnyContent]
 @Singleton
 class AuthenticatedIdentifierAction @Inject()(override val authConnector: AuthConnector,
                                               userAllowListConnector: UserAllowListConnector,
+                                              idGenerator: IdGenerator,
                                               config: FrontendAppConfig,
                                               playBodyParsers: BodyParsers.Default)
                                              (implicit override val executionContext: ExecutionContext)
   extends IdentifierAction with AuthorisedFunctions with Logging {
 
+  private[actions] def handleWithCorrelationId[A](request: Request[A],
+                                                  extraContext: String)
+                                                 (block: RequestWithCorrelationId[A] => Future[Result]): Future[Result] = {
+    val methodLoggingContext: String = "handleWithCorrelationId"
+
+    val infoLogger: String => Unit = infoLog(
+      secondaryContext = methodLoggingContext,
+      extraContext = Some(extraContext)
+    )
+
+    val warnLogger: (String, Option[Throwable]) => Unit = warnLog(
+      secondaryContext = methodLoggingContext,
+      extraContext = Some(extraContext)
+    )
+
+    infoLogger("Attempting to retrieve Correlation ID from request headers")
+
+    val correlationId = request.headers
```

Comment:
This won't work... it will create a new `correlationId` for each page. Better we close this PR and just add/update the    logs. We can refactor the code later if required